tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -7428,7 +7428,7 @@ UINT64 HLSLExternalSource::ScoreCast(QualType pLType, QualType pRType)
 #undef SCORE_COND
 
   // Make sure our scores fit in a UINT64.
-  C_ASSERT(SCORE_MIN_SHIFT + SCORE_PARAM_SHIFT * 8 <= 64);
+  static_assert(SCORE_MIN_SHIFT + SCORE_PARAM_SHIFT * 8 <= 64);
```

Comment:
This is far from the only C_ASSERT in this file let alone the complete source. It is defined in WinAdapter to static_assert. Why was only this location changed?

Reply:
This one hit -Wunused-local-typedef  " unused typedef '__C_ASSERT__'.
not sure why :(

Reply:
The Windows system version of `C_ASSERT` is the old hack to produce a negative sized array typedef. On Windows this won't use the `WinAdapter` version instead it will use the system header version:

```
#define C_ASSERT(e) typedef char __C_ASSERT__[(e)?1:-1]
```
[Source](https://learn.microsoft.com/en-us/windows/win32/api/winnt/nf-winnt-c_assert)